src/main/java/org/molgenis/vcf/decisiontree/filter/ConsequenceAnnotatorImpl.java

```diff
@@ -3,32 +3,63 @@
 import static org.molgenis.vcf.decisiontree.filter.DecisionUtils.getDecisionClass;
 import static org.molgenis.vcf.decisiontree.filter.DecisionUtils.getDecisionLabelsString;
 import static org.molgenis.vcf.decisiontree.filter.DecisionUtils.getDecisionsPath;
+import static org.molgenis.vcf.decisiontree.runner.HeaderAnnotator.*;
+import static org.molgenis.vcf.decisiontree.runner.VepHelper.getVepId;
 
+import htsjdk.variant.vcf.VCFHeader;
 import org.molgenis.vcf.decisiontree.filter.model.Decision;
+import org.molgenis.vcf.decisiontree.filter.model.Field;
+import org.molgenis.vcf.decisiontree.filter.model.FieldType;
+import org.molgenis.vcf.decisiontree.filter.model.NestedField;
+import org.molgenis.vcf.decisiontree.runner.info.NestedHeaderLine;
+import org.molgenis.vcf.decisiontree.runner.info.VepMetadataMapper;
+
+import java.util.Arrays;
 
 public class ConsequenceAnnotatorImpl implements ConsequenceAnnotator {
 
   private final boolean writeLabels;
   private final boolean writePaths;
+  private final VCFHeader annotatedHeader;
+  private final VepMetadataMapper vepMetadataMapper;
 
-  public ConsequenceAnnotatorImpl(boolean writeLabels, boolean writePaths) {
+  public ConsequenceAnnotatorImpl(boolean writeLabels, boolean writePaths, VCFHeader annotatedHeader, VepMetadataMapper vepMetadataMapper) {
     this.writeLabels = writeLabels;
     this.writePaths = writePaths;
+    this.annotatedHeader = annotatedHeader;
+    this.vepMetadataMapper = vepMetadataMapper;
   }
 
   @Override
   public String annotate(Decision decision, String consequence) {
-    StringBuilder csqBuilder = new StringBuilder(consequence);
-    csqBuilder.append("|");
-    csqBuilder.append(getDecisionClass(decision));
+    NestedHeaderLine vepHeaderLine = vepMetadataMapper.map(getVepId(annotatedHeader), annotatedHeader);
```

Comment:
what happens if getVepId returns null?

Reply:
this cannot happen since the metadata is already added and the run would have failed at that point, I'll check it anyway